Updated UB roadmap (#14113)
dotnet/arcade: Documentation/UnifiedBuild/UB-Roadmap-Gantt.md

        
**Workloads PoC**

SDK Workloads vertical build Proof of Concept. Similar to the platform VB PoCs, this effort is focused to identify and uncover all issues related to building the SDK workloads.

**Identify Join Points**

Identification and mapping of the current cross-platform build orchestration.

**Vertical Builds Design**

Leverage the finding of the vertical build PoC works to design vertical builds for each platform properly, with the minimal set of join points.

**Enable Vertical Builds**

Implementation of the vertical builds design with the new se of join points.

**Crossbuilds Design**

Design for cross-arch or cross-platform builds and determine how to define cross-build behavior.

## Source-Build area

**VMR UX Improvements**

Improvements of the VMR user experience. This includes, but is not limited to:
  * Updating source-build to not modify source-controlled files
  * Design of appropriate controls and command line options

>   * Source-build inner clone removal

Reviewer: + incremental build?

**Multi-band SDKs**

> Implementing support for building multiple SDK bands sharing one binary copy of the runtime.

Reviewer: ```suggestion
Implementing support for building multiple SDK bands sharing one binary copy of shared components (e.g. runtime).
```

## Product Validation area
        
**Scenario tests in VMR**

End-to-end scenario tests (such as `dotnet new console` or a running a Stage 2 Linux source-build of the SDK) of the .NET product.

**PR Validation**

Definition and implementation of the set of tests that would be executed as part of the VMR PR validation.

## Product Construction area

**Backflow design**

Design for the backflow from the VMR to the individual product repositories.

**Backflow tooling**

Implementation of the core functionality and CLI tooling for the VMR backflow in to the product repos.

**Dependency Flow Service**

Implementation of the new dependency flow service, that will be extending the current BAR design.

**Maestro Integration**

Integration of the new dependency flow service with Maestro++.

**Dependency Flow Switch Preparation**

Preparation for switching from the existing multi-leayered product dependency flow to the new flat dependency flow between VMR and product repos.

**Depenendency Flow Switch**

> Swith to the new flat dependency flow between VMR and product repos.

Reviewer: ```suggestion
Switch to the new flat dependency flow between VMR and product repos.
```

## Release Infra area

**Release infra investigation & design**

Design for changes necessary to enable releases off of the VMR.

**Signing Design**

Design for signing releases based off of the VMR.

**Identify Repo Dependencies**

Identification of the dependencies between product repos and the layout used to stage the product assets for release.

**Staging / Release Pipeline**

> Updates to the current release infrastructure, namely the staging and release pipelines to be able to base releases both off the current dependency flow for ..NET8 and the VMR for .NET9.

Reviewer: ```suggestion
Updates to the current release infrastructure, namely the staging and release pipelines to be able to base releases both off the current dependency flow for .NET8 and the VMR for .NET9.
```

>     section Common

Reviewer: I'd remove the `Common` lane. There's not much value but just adds clutter.

Author: I want to keep the UB week for now as it adds clearly recognizable starting point for everyone who's been participating on the UB week's discussions.

Author: Unless you (or anyone else) have a strong opinion on this, I'd like to keep it.

Reviewer: Alternatively, RC2 milestone can be UB Week milestone

Author: Let's keep it for now at least for the M2 review to provide clear starting point and then we could remove it.

        Enable Vertical Builds: after net9-p1, 8w
        Crossbuilds Design: after net8-ga, 13w

    section Source-Build
        VMR UX Improvements: after net8-ga, 13w
        Multi-band SDKs: after net9-p4, 4w

    section Product Validation
        Scenario tests in VMR: after net8-rc2, 10w
        PR Validation: after net9-p2, 4w

    section Shared Infra

>         Signing Design: after holiday, 7w

Reviewer: This could go to release infra

Reviewer: Yeah, I am not sure what is "Shared infra"

Author: True, but we also want to implement repo-level signing, which would make it part of the shared tooling. Let's discuss this online.

Author: I've removed the Shared Infra. It was originally meant to represent the DncEng (i.e.,e Matt) area, but as said above, I think this coupling of areas and teams has drawbacks (i.e., raising early questions about funding), so I've changed this to the project areas.

    section Product Construction

>         Backflow tooling: after net8-ga, 6w

Reviewer: There could be a "Backflow design" that has started already as there is a gap in the front of backflow tooling.

        Dependency Flow Service: after holiday, 7w
        Maestro Integration: after net9-p1, 8w

>         Identify Repo Dependencies: after net9-p3, 4w

Reviewer: This goes to release infra?

Author: Correct, thanks
